src/main/kotlin/org/neo4j/graphql/SchemaBuilder.kt

```diff
@@ -99,18 +118,20 @@ object SchemaBuilder {
         return handler
     }
 
-    private fun augmentSchema(sourceSchema: GraphQLSchema, handler: List<AugmentationHandler>, config: SchemaConfig): GraphQLSchema {
+    private fun augmentSchema(sourceSchema: GraphQLSchema, handler: List<AugmentationHandler>): GraphQLSchema {
         val types = sourceSchema.typeMap.toMutableMap()
-        val env = BuildingEnv(types)
-
+        val env = BuildingEnv(types, sourceSchema)
+        val queryTypeName = sourceSchema.queryTypeName()
+        val mutationTypeName = sourceSchema.mutationTypeName()
+        val subscriptionTypeName = sourceSchema.subscriptionTypeName()
         types.values
```

Comment:
in general it would be cool if we could have more transformations than in-place updates

Reply:
I will pay attention on future refactorings

Reply:
Thanks!